Apps/Database/src/Delegates/DBCommunicationDelegate.cs

```diff
@@ -181,5 +216,51 @@ public DBResult<Communication> Delete(Communication communication, bool commit =
             this.logger.LogDebug($"Finished deleting Communication in DB");
             return result;
         }
+
+        private List<Communication> GetActiveBannerCommunications()
+        {
+            List<Communication> communication = this.dbContext.Communication
+                 .OrderByDescending(c => c.CreatedDateTime)
+                 .Where(c => c.CommunicationTypeCode == CommunicationType.Banner
+                     && c.CommunicationStatusCode == CommunicationStatus.New && DateTime.UtcNow <= c.ExpiryDateTime)
+                 .ToList();
+
+            return communication;
+        }
+
+        private bool IsCommunicationValidCommit(Communication communication)
+        {
+            List<Communication> activeBannerCommunication = this.GetActiveBannerCommunications();
+
+            if (communication.CommunicationStatusCode == CommunicationStatus.New)
+            {
+                if (activeBannerCommunication != null && activeBannerCommunication.Count == 0)
+                {
+                    return true;
+                }
+                else
+                {
+                    var validCommunications = activeBannerCommunication
+                            .OrderByDescending(c => c.EffectiveDateTime)
```

Comment:
A list is being returned but only count is being used.  Is it necessary to still have an OrderByDescending clause?  Order by is an added cost for SQL performance.

Reply:
@BrianMaki  I used the list because I want to use the recent effective date to show in the message but I am still waiting for Nino feedback on UI , if he wants to show that information.